### DISCUSSION TITLE
How to draw a color-filled irregular shape

### DISCUSSION THREAD

**Discussion by Wu-Haitao** [raysan5/raylib#2478] [Q&A] [answered]
Just curious if there is a way of drawing a color-filled irregular shape?
Something like this: https://docs.microsoft.com/en-us/dotnet/api/system.drawing.graphics.fillclosedcurve

**Comment by chriscamacho** [comment, marked as answer]
void DrawTexturePoly(Texture2D texture, Vector2 center, Vector2 *points, Vector2 *texcoords, int pointsCount, Color tint);      // Draw a textured polygon

the down side of this is that it is only suitable for shapes where there is a unobstructed line between the centre points and the points defining the polygon, theres a name for that sort of poly but I can't bring it to mind!

**Reply by Wu-Haitao** [reply]
Yes this is exactly what I need. Thank you!

**Reply by waterstart** [reply]
convex polygons for people visiting this thread
